### PR TITLE
Update redux-saga

### PR DESCRIPTION
**Pull request by chrisnojima**
see https://github.com/redux-saga/redux-saga/releases/tag/v1.0.0 for relevant changes to the api

### CODE REVIEW

File: shared/actions/chat2/index.tsx [hunk 1]
```diff
@@ -141,7 +141,13 @@ function* requestMeta(state: TypedState, _: Chat2Gen.MetaHandleQueuePayload) {
     : []
   const unboxSomeMoreActions = metaQueue.size ? [Saga.put(Chat2Gen.createMetaHandleQueue())] : []
   const delayBeforeUnboxingMoreActions =
-    toUnboxActions.length && unboxSomeMoreActions.length ? [Saga.callUntyped(Saga.delay, 100)] : []
+    toUnboxActions.length && unboxSomeMoreActions.length
+      ? [
+          Saga.callUntyped(function*() {
+            yield Saga.delay(100)
```
Comment (chrisnojima) [inline shared/actions/chat2/index.tsx]: delay is an effect now

File: shared/actions/platform-specific/index.native.tsx [hunk 2]
```diff
@@ -329,7 +329,7 @@ function* loadStartupDetails() {
   })
   const linkTask = yield Saga._fork(Linking.getInitialURL)
   const initialPush = yield Saga._fork(getStartupDetailsFromInitialPush)
-  const [routeState, link, push] = yield Saga.join(routeStateTask, linkTask, initialPush)
+  const [routeState, link, push] = yield Saga.join([routeStateTask, linkTask, initialPush])
```
Comment (chrisnojima) [inline shared/actions/platform-specific/index.native.tsx]: join takes an array now

File: shared/local-debug.d.ts [hunk 3]
```diff
@@ -15,8 +15,6 @@ export declare const printRPC: boolean
 export declare const printRPCBytes: boolean
 export declare const printRPCWaitingSession: boolean
 export declare const printRPCStats: boolean
-export declare const reduxSagaLogger: boolean
```
Comment (chrisnojima) [inline shared/local-debug.d.ts]: removing the redux saga logger as i'm likely the only one who really used it and they split it into a different package and the api it uses changed so its not worth fixing right now. if we need it we can look again

File: shared/actions/chat2/index.tsx (hunk 1 above)
Comment (songgao) [reply shared/actions/chat2/index.tsx]: would this work?

```
toUnboxActions.length && unboxSomeMoreActions.length ? [ Saga.delay(100) ] : []
```

File: shared/local-debug.d.ts (hunk 3 above)
Comment (songgao) [reply shared/local-debug.d.ts]: Is this for logging the actions (like `FsGen.folderListLoad`)? I find them useful sometimes.

Comment (chrisnojima) [reply shared/local-debug.d.ts]: no, its for logging the internals of saga itself, likely you've never used it

File: shared/actions/chat2/index.tsx (hunk 1 above)
Comment (chrisnojima) [reply shared/actions/chat2/index.tsx]: yes! good call

**Pull request merged**